spec/std/process_spec.cr

```diff
@@ -379,6 +387,107 @@ describe Process do
           end
         end
       {% end %}
+
+      if {{ flag?(:win32) }}
+        it "finds binary in parent `$PATH`, not `env`" do
+          Process.run(*print_env_command, env: {"PATH" => ""})
+        end
+      else
+        # FIXME: This behaviour is incorrect. It should lookup the command in
+        # the parent process' `$PATH`, without any changes from `env`.
+        # https://github.com/crystal-lang/crystal/issues/6464#issuecomment-3391000914
+        it "finds binary in `env`" do
+          expect_raises(File::NotFoundError) do
+            Process.run(*print_env_command, env: {"PATH" => ""})
+          end
+        end
+      end
+
+      it "errors on invalid key" do
+        {% begin %}
+          expect_raises({% if flag?(:win32) %}ArgumentError, "Invalid env key" {% else %} RuntimeError, "Invalid argument"{% end %}) do
+            Process.run(*print_env_command, env: {"" => "baz"})
+          end
+          expect_raises({% if flag?(:win32) %}ArgumentError, "Invalid env key" {% else %} RuntimeError, "Invalid argument"{% end %}) do
+            Process.run(*print_env_command, env: {"foo=bar" => "baz"})
+          end
+        {% end %}
+      end
+
+      it "errors on zero char in key" do
+        expect_raises({{ flag?(:win32) }} ? ArgumentError : RuntimeError, "String `key` contains null byte") do
+          Process.run(*print_env_command, env: {"foo\0" => "baz"})
+        end
+      end
+
+      it "errors on zero char in value" do
+        expect_raises({{ flag?(:win32) }} ? ArgumentError : RuntimeError, "String `value` contains null byte") do
+          Process.run(*print_env_command, env: {"foo" => "baz\0"})
+        end
+      end
+    end
+
+    it "errors with empty command" do
+      {% begin %}
+        expect_raises({% if flag?(:win32) %} IO::Error, "The parameter is incorrect" {% else %} File::NotFoundError{% end %}) do
+          Process.run("")
+        end
+      {% end %}
+    end
+
+    it "errors with too long command" do
+      pending! unless {{ flag?(:linux) }}
+
+      path_max = {% if LibC.has_constant?(:PATH_MAX) %}
+                   LibC::PATH_MAX
+                 {% else %}
+                   10_000
+                 {% end %}
+
+      expect_raises(IO::Error, /File ?name too long/) do
+        Process.run("a" * (path_max + 1))
+      end
+
+      # The pathname itself is not too long, but it will be when combined with
+      # any path prefix.
+      expect_raises(IO::Error, /File ?name too long/) do
+        Process.run("a" * path_max)
+      end
+    end
+
+    describe "$PATH" do
+      it "works with unset $PATH" do
+        with_env("PATH": nil) do
+          Process.run(*path_search_command)
+        end
+      end
+
+      it "errors with empty $PATH" do
+        pending! if {{ flag?(:win32) }}
```

Comment:
I'm not quite sure why Windows still finds `cmd.exe` despite empty `PATH`.

When I set `PATH=""` in powershell, it fails as expected:
```console
$ $env:PATH=""
$ cmd.exe
cmd.exe: The term 'cmd.exe' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```